src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] dataArray = readFile(fromFileName);
+        String calculations = calculate(dataArray);
+        writeFile(toFileName, calculations);
+    }
+
+    public String[] readFile(String fromFileName) {
+        Path fromFilePath = Paths.get(fromFileName);
+        String[] dataArray;
+        try {
```

Comment:
use try catch with resources, e.g.
```
  try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filename))) {
  ```

Reply:
But NIO does not need try catch with resources. Should I use io?